doc/reference.md

```diff
@@ -272,6 +273,26 @@ Some examples are as follows:
 | `{ x <- f32le, y <- f32le }`               | `{ x : F32, y : F32 }`                 |
 | `{ len <- u16be, data <- array16 len s8 }` | `{ len : U16, data : Array16 len S8 }` |
 
+### Conditional formats
+
+Conditional formats are formats with an associated condition predicate. The format
+will only succeed if the condition is `true`. This can be used to verify magic numbers
+or version expectations. E.g.
+
+```fathom
+{ magic <- { magic <- u64le | (u64_eq magic 0x00ffffffffffff00) } }
+```
+
+```fathom
+{ major_version <- { version <- u16be | (u16_lte version 2) } }
+```
+
+#### Representation of conditional formats
+
+The [representation](#format-representations) of a conditional format is the same
+as the format without the condition when the condition evaluates to `true`. If
+the condition is `false` then it's equivalent to the [fail format](#fail-format).
```

Comment:
I'm not sure this is entirely accurate… this kind of implies something like `Repr format | Void`? Where as we want something more like:

| format                    | `Repr` format         |
| ------------------ | ------------------- |
| `{ x <- format \| cond }` | `Repr format` |

Not sure how to express this better in words, but it might be more accurate to say that “the representation of a conditional format is the same as the representation of the format guarded by the predicate”. It's a bit clunky to describe in words as we don't really have a name for what we're talking about. 😕

Reply:
> “the representation of a conditional format is the same as the representation of the format guarded by the predicate”

Happy to change it to this. It's basically what I was was trying to say 😅